src/main/xerus/monstercat/api/Playlist.kt

```diff
@@ -61,9 +76,13 @@ object Playlist {
 	
 	fun setTracks(playlist: Collection<Track>) {
 		history.clear()
-		tracks.setAll(playlist)
+		val checkedTracks = if(SKIPUNLICENSABLE()) removeUnsafe(playlist) else playlist
+		if(checkedTracks.isEmpty()) showUnsafeAlert()
+		tracks.setAll(checkedTracks)
 	}
 	
+	private fun removeUnsafe(tracks: Collection<Track>) = tracks.filter { it.licensable && it.artistsTitle != "" && it.artistsTitle != "Monstercat" }
```

Comment:
don't spread out related methods

Reply:
Please explain ?

Reply:
1.  This removeUnsafe function is placed a bit randomly. The methods about licensability should be grouped logically.
2. You haven't adjust safe to licensable. If I leave such comments as the previous, please be more thorough.

Reply:
Fixed in 38fa4685854ee6ec11e55993a63f0e10a9243fa9 plus docs.

Reply:
OOPS here e105e2d65ffbd67dc3e2d59eeb6efae4f83d0fa9